File: src/main/java/net/mreunionlabs/tapgradle/generate/util/PackageDirectoryCreator.java
package net.mreunionlabs.tapgradle.generate.util;

import java.io.File;

/**
 * Created by abangkis on 6/16/2016.
 */
public class PackageDirectoryCreator {
    public static File createPackageDir(File root, String packageString) {
        String[] split = packageString.split("\\.");

        File packageDir = root;
        for (String dir : split) {
            if (dir != null && !"".equals(dir)) {
                packageDir = new File(packageDir, dir);
                packageDir.mkdir();
            }
        }
        return packageDir;
    }
}

File: src/main/java/net/mreunionlabs/tapgradle/generate/util/PackageDirectoryCreator.kt
package net.mreunionlabs.tapgradle.generate.util

import java.io.File

/**
 * Created by abangkis on 6/16/2016.
 */
object PackageDirectoryCreator {
    fun createPackageDir(root: File, packageString: String): File {
        val split = packageString.split("\\.".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()

        var packageDir = root
        for (dir in split) {
            if ("" != dir) {
                packageDir = File(packageDir, dir)
                packageDir.mkdir()
            }
        }
        return packageDir
    }
}
